src/names.c

```diff
@@ -34,27 +34,35 @@ SEXP vec_as_names(SEXP names, const struct name_repair_opts* opts) {
   case name_repair_minimal: return vctrs_as_minimal_names(names);
   case name_repair_unique: return vec_as_unique_names(names, opts->quiet);
   case name_repair_universal: return vec_as_universal_names(names, opts->quiet);
-  case name_repair_check_unique: return vec_validate_unique_names(names, opts->arg);
+  case name_repair_check_unique: return check_unique_names(names, opts);
   case name_repair_custom: return vec_as_custom_names(names, opts);
   }
   never_reached("vec_as_names");
 }
 
 // [[ register() ]]
-SEXP vctrs_as_names(SEXP names, SEXP repair, SEXP repair_arg, SEXP quiet) {
-  if (!r_is_bool(quiet)) {
-    Rf_errorcall(R_NilValue, "`quiet` must a boolean value.");
+r_obj* ffi_as_names(r_obj* names,
+                    r_obj* repair,
+                    r_obj* ffi_repair_arg,
+                    r_obj* ffi_quiet,
```

Comment:
Neat naming convention idea, since you have `bool quiet` below and that is the "better" name